### ISSUE TITLE
[Bug] Garbage output from StableLM and Pythia

### ISSUE THREAD

**Issue by tienthanhdhcn** [microsoft/DeepSpeed-MII#194]
**Problem**
I am using DeepSpeed-MII for serving StableLM ([OpenAssistant/stablelm-7b-sft-v7-epoch-3](https://huggingface.co/OpenAssistant/stablelm-7b-sft-v7-epoch-3)) and Pythia ([OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5](https://huggingface.co/OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5))

However, it generates some garbage texts which are completely different from using HF with the same settings.

**Reproduce**
- Latest codebase, A10 GPUs
- Using StableLM ([OpenAssistant/stablelm-7b-sft-v7-epoch-3](https://huggingface.co/OpenAssistant/stablelm-7b-sft-v7-epoch-3))  or Pythia ([OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5](https://huggingface.co/OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5))
- mii_config: `{"tensor_parallel": 4, "dtype": "fp16", "load_with_sys_mem": True, "port_number": 8421, "skip_model_check": True, "max_tokens": 2048}`

**Comment by satpalsr** [comment]
@tienthanhdhcn  yeah the same problem is reported at multiple places in deepspeed repo too. Same can be verified with this [script](https://github.com/cr458/gpt-neox/blob/deepspeed-bench/inference/benchmark.py).

**Comment by mrwyattii** [comment]
@tienthanhdhcn could you please share the output that you are seeing? I just tested `OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5` with DeepSpeed@master and DeepSpeed-MII@main and here is some of the output I see:
`"Seattle is a city of contradictions. It is a vibrant, diverse metropolis, yet it"`

This matches the output when using `transformers.pipeline`:
`"Seattle is a city of contradictions. It is a vibrant, diverse metropolis, yet it"`

Here is the script I used:
```python
import mii
import torch
from transformers import pipeline

mii_configs = {"tensor_parallel": 1, "dtype": "fp16", "skip_model_check":True}
mii.deploy(task='text-generation',
           model="OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5",
           deployment_name="test_deployment",
           mii_config=mii_configs)

generator = mii.mii_query_handle("test_deployment")
mii_result = generator.query({'query': ["Seattle is"]}, do_sample=False)
mii.terminate("test_deployment")

generator = pipeline("text-generation", "OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5", device=0, torch_dtype=torch.half)
hf_result = generator(["Seattle is"], do_sample=False)

print("MII:")
print(mii_result.response)
print("HF:")
print(hf_result)
```
